safenode/src/network/event.rs

```diff
@@ -107,7 +110,7 @@ impl SwarmDriver {
                     let new_peers: HashSet<PeerId> =
                         closest_peers.peers.clone().into_iter().collect();
                     current_closest.extend(new_peers);
-                    if current_closest.len() >= usize::from(K_VALUE) || step.last {
+                    if current_closest.len() >= CLOSE_GROUP_SIZE || step.last {
```

Comment:
I wonder about this...
If we haven't got all from the bucket, we can't really be sure we actually have _the closest_ ones. Only that we have the number of peers that we want to have _if they are the closest_.

Reply:
What I mean is, that we're not ensured to get these results in any particular order. So when we have the n closest, those might just as well be the _farthest_ oners out of the bucket.

Reply:
>So when we have the n closest, those might just as well be the farthest ones out of the bucket, but we didn't yet receive the others, so we don't know.
The event `OutboundQueryProgressed` returns the final result, and there has been some incomplete docs/naming in libp2p? Added more details in the below comment.

Reply:
Yes, I'd prefer stay with the current K_VALUE

Reply:
Reverting this change as per the Slack discussion